server/src/services/semanticHighlight/highlighters/FunctionDefinitionHighlighter.ts

```diff
@@ -0,0 +1,21 @@
+import { SemanticTokenTypes } from "vscode-languageserver-protocol";
+import { NodeType } from "@nomicfoundation/slang/cst";
+import { RuleKind, TokenKind } from "@nomicfoundation/slang/kinds";
+import { HighlightVisitor } from "../HighlightVisitor";
+import { SlangNodeWrapper } from "../../../parser/slangHelpers";
+
+// Highlights function definitions
+export class FunctionDefinitionHighlighter extends HighlightVisitor {
+  public tokenKinds = new Set([TokenKind.Identifier]);
```

Comment:
Since most of the CST is identifiers, and very few are function/contract/struct/etc... definitions, I wonder if the cursor should be using the same logic as symbols, by searching by the parent kind instead. Thoughts?

Reply:
I just tested and after removing keyword highlighters (because they are in the textmate grammar), highlighting is very fast. It takes around ~6ms on bigger contracts (excluding parsing). So I don't think we need to change this for now

Reply:
Great!